### PR TITLE
Upload image for Docker Pull workflow

### PR DESCRIPTION
**Pull request by corrieb**
[skip ci]

### CODE REVIEW

**Comment by caglar10ur** [comment]
Step 4 should be between imagec and hub

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by jzt** [comment]
Minor: it might make more sense to call them "layers" in step 7, since this is taking place in docker world and not portlayer world.

LGTM

**Comment by corrieb** [comment]
@jzt I think I'll keep images. Not every container implementation uses layers, so let's keep it ambiguous

**Comment by corrieb** [comment]
@caglar10ur @jzt I added a design doc that references the image to the PR. Can you give that a check over. It's not complete yet, but it's a good start

File: doc/design/image-management.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+A big part of a container runtime is how it uses and manages filesystem images. There are various approaches to managing container images, but the net result needs to be enough binaries to run the main container process. Those binaries may infact be a single statically-linked binary, or it could be a operating system dependencies, third party libraries etc.
+
+Given that each containerVM in VIC does not share a filesystem with anything else, for a container VM to be functional, it requires the following:
+* OS binaries from which to boot. These are contained in the ISO attached to the containerVM
+* A disk, representing the container filesystem, mounted at "/". The disk may contain OS libraries, but must contain the binaries corresponding to the entry point of the container. The disk will typically be R/W and the underlying implementation may capture all reads into a layer which can be committed into a snapshot
+* 0 or more R/W disks which persist beyond the lifespan of the container
+
+#### Port Layer Storage
+
+#### Imagec
```
Comment (hickeng) [inline doc/design/image-management.md]: Add a link to/from doc/design/components.md#imagec and consider whether that description needs updating.

File: doc/design/image-management.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+### How VIC manages filesystem images
+
+A big part of a container runtime is how it uses and manages filesystem images. There are various approaches to managing container images, but the net result needs to be enough binaries to run the main container process. Those binaries may infact be a single statically-linked binary, or it could be a operating system dependencies, third party libraries etc.
```
Comment (jzt) [inline doc/design/image-management.md]: s/it could be a/could consist of

File: doc/design/image-management.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Given the out-of-band nature of this transaction, plus the potential for the exchange of image data to be be automated or even managed by a different control plane, it makes sense to provide a binary to allow for this. This binary is imagec https://github.com/corrieb/vic/blob/master/cmd/imagec. 
+
+#### Image metadata
+
+Image metadata can originate from two main sources. From the creator of the image and from the user of the image. Metadata originating from the creator will likely be in some kind of image maninfest and metadata originating from the user will be contextual data about the image, such as a user-defined tag. The scope of this metadata differs, in that the image metadata should be immutable and shared between VCHs, whereas the user-defined metadata will be scoped to a VCH. Both types of data should be persisted
```
Comment (hickeng) [inline doc/design/image-management.md]: The creator pushes the image to a repo with a tag - in fact tagging an image is the only way in docker to push to a repo other than hub.
I think this breakdown isn't so much creator vs user as it is audit data about the image creation (which should obviously be immutable) and data about image publishing (names, tags, etc) which don't have any impact on the actual image content.

Comment (jzt) [inline doc/design/image-management.md]: s/maninfest/manifest

Comment (jzt) [inline doc/design/image-management.md]: `Both types of data should be persisted` - this sentence ends abruptly.. not sure if you meant to add more or just forgot a period. :)

File: doc/design/image-management.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+Given the out-of-band nature of this transaction, plus the potential for the exchange of image data to be be automated or even managed by a different control plane, it makes sense to provide a binary to allow for this. This binary is imagec https://github.com/corrieb/vic/blob/master/cmd/imagec. 
+
+#### Image metadata
+
+Image metadata can originate from two main sources. From the creator of the image and from the user of the image. Metadata originating from the creator will likely be in some kind of image maninfest and metadata originating from the user will be contextual data about the image, such as a user-defined tag. The scope of this metadata differs, in that the image metadata should be immutable and shared between VCHs, whereas the user-defined metadata will be scoped to a VCH. Both types of data should be persisted 
+
+#### High-level control flow
+
```
Comment (hickeng) [inline doc/design/image-management.md]: did you want to cover how the data is persisted and why the separation, or is this doc just for the high level portion?

File: doc/design/image-management.md (hunk 1 above)
Comment (corrieb) [inline doc/design/image-management.md]: I've added better links. I don't think the components.md needs updating for now

File: doc/design/image-management.md (hunk 3 above)
Comment (corrieb) [inline doc/design/image-management.md]: Good distinction George. Improved it.

File: doc/design/image-management.md (hunk 4 above)
Comment (corrieb) [inline doc/design/image-management.md]: This is just high level for now. I basically wanted something to add context to the diagram.

**Comment by hickeng** [comment]
LGTM

**Comment by corrieb** [comment]
Rebased

**Pull request merged**